kafka-client/src/main/java/dev/responsive/kafka/internal/db/FactWriterFactory.java

```diff
@@ -17,30 +17,62 @@
 package dev.responsive.kafka.internal.db;
 
 import com.datastax.oss.driver.api.core.cql.BoundStatement;
-import dev.responsive.kafka.internal.utils.SessionClients;
+import dev.responsive.kafka.internal.db.partitioning.ResponsivePartitioner;
+import dev.responsive.kafka.internal.db.partitioning.ResponsivePartitioner.DefaultPartitioner;
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
 
-public class FactWriterFactory<K> implements WriterFactory<K> {
+public class FactWriterFactory<K> extends WriterFactory<K, Integer> {
 
-  private final RemoteTable<K, BoundStatement> table;
+  private final RemoteTable<K, Integer, BoundStatement> table;
+  private final CassandraClient client;
+  private final ResponsivePartitioner<K, Integer> partitioner;
+  private final int kafkaPartition;
 
-  public FactWriterFactory(final RemoteTable<K, BoundStatement> table) {
+  public FactWriterFactory(
+      final RemoteTable<K, Integer, BoundStatement> table,
+      final CassandraClient client,
+      final int kafkaPartition
+  ) {
+    super(String.format("FactWriterFactory [%s-%d] ", table.name(), kafkaPartition));
     this.table = table;
+    this.client = client;
+    this.partitioner = new DefaultPartitioner<>();
+    this.kafkaPartition = kafkaPartition;
   }
 
   @Override
-  public RemoteWriter<K> createWriter(
-      final SessionClients client,
-      final int partition
+  public RemoteWriter<K, Integer> createWriter(
+      final Integer tablePartition
   ) {
     return new FactSchemaWriter<>(
-        client.cassandraClient(),
+        client,
         table,
-        partition
+        tablePartition
     );
   }
 
   @Override
-  public String toString() {
-    return "FactWriterFactory{}";
+  public String tableName() {
+    return table.name();
+  }
+
+  @Override
+  public Integer tablePartitionForKey(final K key) {
+    return partitioner.tablePartition(kafkaPartition, key);
   }
+
+  @Override
+  public RemoteWriteResult<Integer> setOffset(final long offset) {
```

Comment:
it feels weird to me that writer factory is directly doing "things" outside of creating a writer?

Reply:
Yes it very much needs to be renamed. I absolutely plan to change it, but wanted to leave that for a followup PR just to cut down on the total number of line changes here (which are already out of control). I'll do that immediately after this is merged